synapse/cores/common.py

```diff
@@ -125,38 +126,11 @@ def __init__(self, link):
 
         self._initCortex()
 
-        # FIXME unicode / "word" characters
-        #self.addSubType('syn:tag','str', regex='^[a-z0-9._]+$', lower=1)
-        #self.addSubType('syn:prop','str', regex='^[a-z0-9:_]+$', lower=1)
-        #self.addSubType('syn:type','str', regex='^[a-z0-9:_]+$', lower=1)
-
-        self.addTufoForm('syn:tag', ptype='syn:tag')
-        self.addTufoProp('syn:tag','up',ptype='syn:tag')
-        self.addTufoProp('syn:tag','doc',defval='',ptype='str')
-        self.addTufoProp('syn:tag','depth',defval=0,ptype='int')
-        self.addTufoProp('syn:tag','title',defval='',ptype='str')
-
-        self.addTufoForm('syn:model',ptype='syn:prop', doc='prefix for all forms within the model')
-        self.addTufoForm('syn:model:version', ptype='int', doc='model version for the model loaded in the cortex')
-
-        self.addTufoForm('syn:type',ptype='syn:type')
-        self.addTufoProp('syn:type','doc',ptype='str', defval='??', doc='Description for this type')
-        self.addTufoProp('syn:type','ver',ptype='int', defval=1, doc='What version is this type')
-        self.addTufoProp('syn:type','base',ptype='str', doc='what type does this type extend?', req=True)
-        self.addTufoGlob('syn:type','info:*')
-
-        self.addTufoForm('syn:form',ptype='syn:prop')
-        self.addTufoProp('syn:form','doc',ptype='str', doc='basic form definition')
-        self.addTufoProp('syn:form','ver',ptype='int', doc='form version within the model')
-        self.addTufoProp('syn:form','model',ptype='str', doc='which model defines a given form')
+        self.addDefaultTypes()
+        self.bootstrapForms()
 
         # TODO: syn:err with rate limiting?
 
-        self.addTufoForm('syn:prop',ptype='syn:prop')
-        self.addTufoProp('syn:prop','doc',ptype='str')
-        self.addTufoProp('syn:prop','form',ptype='syn:prop')
-        self.addTufoProp('syn:prop','ptype',ptype='syn:type')
-
```

Comment:
all these forms are conceptually part of the data model, so they are moved into `synapse.datamodel`.